apps/analysis/serializers.py

```diff
@@ -52,6 +54,25 @@ def validate(self, data):
         return data
 
 
+class DiscardedEntrySerializer(serializers.ModelSerializer):
+    tag_display = serializers.CharField(source='get_tag_display', read_only=True)
+
+    class Meta:
+        model = DiscardedEntry
+        fields = '__all__'
+
+    def validate(self, data):
+        if data['analysis_pillar'].id != int(self.context['view'].kwargs.get('analysis_pillar_id', None)):
+            raise serializers.ValidationError(
+                f'Analysis pillar doesnot match'
+            )
```

Comment:
We were directly using analysis_pillar_id, right? Why are we asking this from the client and validating it?

Reply:
During validation we cannot get the `analysis_pillar` from the url kwargs since we were passing the `analysis_pillar_id`, we require `analysis_pillar` project for the validation.

Reply:
We can get the analysis_pillar using analysis_pillar_id. There is no need to ask for id in both URL and body from the client.

Reply:
Done `8d74dae`